src/common/logger.ts

```diff
@@ -0,0 +1,165 @@
+			CONSOLE_LOG.apply( console, [ 'Flushing log file of length ' + this._logLines.join( '\n' ).length + ' with ' + this._logLines.length + ' rows' ] );
+			this._file = fs.createWriteStream( this._filePath, {
+				flags: 'a',
+				encoding: 'utf8',
+			} );
+		}
+		catch ( err ) {
+			CONSOLE_LOG.apply( console, [ 'Babel sucks: ' + err.message + '\n' + err.stack ] );
+		}
+	}
+
+	private static _log( ...args: any[] )
+	{
+		CONSOLE_LOG.apply( console, args );
+		let str = util.format.apply( console, args ).split( '\n' );
+		for ( let strVal of str ) {
+			this._logLines.push( strVal );
+		}
+		if ( this._file ) {
+			this._file.write( str + '\n' );
+		}
+		if ( this._logLines.length > LOG_LINES ) {
+			this._logLines = _.clone( this._logLines.slice( this._logLines.length - LOG_LINES ) );
+		}
+	}
+
+	private static _logErr( ...args: any[] )
+	{
+		CONSOLE_ERR.apply( console, args );
+		let str = util.format.apply( console, args ).split( '\n' );
+		for ( let strVal of str ) {
+			this._logLines.push( strVal );
+		}
+		if ( this._file ) {
+			this._file.write( str + '\n' );
+		}
+		if ( this._logLines.length > LOG_LINES ) {
+			this._logLines = this._logLines.slice( this._logLines.length - LOG_LINES );
+		}
+	}
+
+	static async hijack( file?: string)
+	{
+		if ( this._hijacked ) {
+			return;
+		}
+
+		console.log( 'Hijacking console' );
+		this._filePath = file || 'client.log';
+		if ( await Common.fsExists( this._filePath ) ) {
+			try {
+				let readLines = await Common.fsReadFile( this._filePath, 'utf8' );
+				console.log( typeof readLines );
+				this._logLines = readLines.split( '\n' );
+				if ( this._logLines.length > LOG_LINES ) {
+					this._logLines = this._logLines.slice( this._logLines.length - LOG_LINES );
+				}
+			}
+			catch ( err ) {
+				console.log( 'Seroiusly babel gtfo: ' + err.message + '\n' + err.stack );
+			}
+		}
+		this._file = fs.createWriteStream( this._filePath, {
+			flags: 'a',
+			encoding: 'utf8',
+		} );
+		let flushFunc: ( ...args:any[] ) => void = this._flushFile.bind( this );
+		this._flushInterval = setInterval( flushFunc, 10000 );
+
+		console.log( 'Hijacked console, mmhm. mmhm.')
+		console.log = this._log.bind( this );
+		console.info = this._log.bind( this );
+		console.warn = this._logErr.bind( this );
+		console.error = this._logErr.bind( this );
+
+		this._hijacked = true;
+	}
+
+	static async unhijack()
+	{
+		if ( !this._hijacked ) {
+			return;
+		}
+
+		CONSOLE_LOG.apply( console, [ 'Unhijacking console' ] );
+		clearInterval( this._flushInterval );
+		if ( this._file ) {
+			this._file.close();
+		}
+		await Common.fsWriteFile( this._filePath, this._logLines.join( '\n' ) );
+
+		CONSOLE_LOG.apply( console, [ 'Unhijacked console' ] );
+		console.log = CONSOLE_LOG;
+		console.info = CONSOLE_LOG;
+		console.warn = CONSOLE_ERR;
+		console.error = CONSOLE_ERR;
+
+		this._hijacked = false;
+	}
+
+	static getClientLog( title: string, description?: string ): IClientLog
```

Comment:
This shouldn't take a title, description. There's no point. It just returns them back.